src/preferences/device.js

```diff
@@ -611,6 +613,15 @@ var Panel = GObject.registerClass({
         }
     }
 
+    _setCustomChargeLevel(spin) {
+        const settings = this.pluginSettings('battery');
+        const oldLevel = settings.get_value('custom-battery-notification-value').unpack();
+        const newLevel = GLib.Variant.new('d', spin.get_value());
+
+        if (newLevel !== oldLevel)
```

Comment:
Here you're trying to compare two `GVariant` objects as though they're literals. If you want to check the equality of variants you should do use [`GLib.Variant.equal()`](https://gjs-docs.gnome.org/glib20/glib.variant#method-equal) like `if (!newLevel.equal(oldLevel)`.

There's not much point in creating intermediate GVariant objects in JavaScript though. You should probably just do:

```js
let oldLevel = settings.get_double('custom-battery-notification-value');
let newLevel = spin.get_value();

if (newLevel !== oldLevel)
    settings.set_double('custom-battery-notification-value', newValue);
```

You should also have a second method like `_getCustomeChargeLevel()` that is called when the widget is constructed. I think the simpler way is just to use a property binding, but that's up to you.